### ISSUE TITLE
Some question about the code 

### ISSUE THREAD

**Issue by burness** [buriburisuri/ac-gan#2]
Hi @buriburisuri , your code [ac-gan loss](https://github.com/buriburisuri/ac-gan/blob/master/train.py#L141), Here the loss which need to be optimized seems not be a 0d tensor, didn't it need to sg_mean() with the loss_c_r and loss_c_d?

**Comment by buriburisuri** [comment]
@burness  You are right.  Strictly, sg_mean() is needed, but tensorflow optimizer will do averaging implicitly. So, for convenience,  it's general to omit reduce_mean() at final loss.  Thanks.

**Comment by burness** [comment]
@buriburisuri  Thanks! Get it
